bgs/bgs.go

```diff
@@ -941,7 +943,9 @@ func (bgs *BGS) handleFedEvent(ctx context.Context, host *models.PDS, env *event
 			return fmt.Errorf("rebase was true in event seq:%d,host:%s", evt.Seq, host.Host)
 		}
 
-		if host.ID != u.PDS && u.PDS != 0 {
+		if host.RelayAllowed {
+			// don't check that source is canonical PDS, allow intermediate relays
+		} else if host.ID != u.PDS && u.PDS != 0 {
```

Comment:
What are the implications here of skipping the `createExternalUser` bit? Do we end up not creating a row for users that are being echoed by a Relay?

Reply:
yeah
![image](https://github.com/user-attachments/assets/ee826e45-eac7-48a0-b0a6-20add58a0a8c)

Reply:
this is a little weird, as it turns out that narelay _does_ keep a little bit of per-user state so that it can emit a warning if there's a discontinuity in a user's feed ( See #915 )

Reply:
I think we'd still probably want to keep track of the users being proxied by a Relay so we can ban them and/or change their upstream status and/or count them right?

Reply:
mehhhh, ok, this needs some more consideration, there's _some_ useful stuff inside `.createExternalUser()` and I can't just throw out the path of code that was under "user from weird place, wat"